skfem/io/meshio.py

```diff
@@ -70,21 +95,17 @@ def from_meshio(m, force_mesh_type=None):
     mtmp = mesh_type(p, t)
 
     try:
-        # element to boundary element type mapping
-        bnd_type = {
-            'line': 'vertex',
-            'triangle': 'line',
-            'quad': 'line',
-            'tetra': 'triangle',
-            'hexahedron': 'quad',
-        }[meshio_type]
+        bnd_type = BOUNDARY_TYPE_MAPPING[meshio_type]
 
         def find_tagname(tag):
             for key in m.field_data:
                 if m.field_data[key][0] == tag:
                     return key
             return None
 
+        if int_data_to_sets:
+            m.int_data_to_sets()
```

Comment:
AFAIC losing of the names is a bug of meshio and should be fixed there. I think the logic in `int_data_to_sets` which tries to match the tag names from the combined `cell_data` tag (`left`, `right` gets transformed to `left-right` in `sets_to_int_data` and the latter is again parsed by `int_data_to_sets`) is not correct.

Reply:
Also, the additional tags with `gmsh` prefix come from the `gmsh22` reader. I think these readers can pretty much arbitrarily add additional stuff.

Reply:
I know, and mysterious new ones keep appearing, like "gmsh:dim_tags" in nschloe/meshio#922.

Reply:
Tagging is now fixed in nschloe/meshio main branch.

Reply:
Good one. Does that fix nschloe/meshio#736?

Reply:
I think partially. It's not reading field_data though. Maybe the docstring should read cell_data instead.

Reply:
The [docstring](https://github.com/nschloe/meshio/blob/b3630c2d8d7649b14d26665ca75679258b6fc483/src/meshio/_mesh.py#L342) seems to have stopped referring to `field_data` a long time ago (I failed to bisect this over the move to the `src` layout nschloe/meshio#1090, but no matter).

I'm not sure what you mean by

> Tagging is now fixed 

When I looked again at nschloe/meshio#736 this morning, I found that the keys to `cell_sets` after `int_data_to_sets` in the MWE there looked like `"set1", "set2", ...` which is different from yesterday, but still a little inscrutable.

One could use something like
```python
{k: f"set{v[0]}" for k, v in mio.field_data.items()}
```
to decode the otherwise opaque names, connecting them back to the ones meaningful in the application?  But I'm not quite clear on the kind of interoperation that is planned in the motivating application.

Reply:
It won't parse `gmsh22` tags from `field_data` because that's not the purpose of `int_data_to_sets`. But the issue here is not related because `gmsh22` writer neither does write the tags to `field_data`. See https://github.com/nschloe/meshio/issues/1160 https://github.com/nschloe/meshio/issues/1164 https://github.com/nschloe/meshio/issues/1165

Reply:
I think tags being in `field_data` is specific to `gmsh22` whereas `int_data_to_sets` tries to be generic w.r.t the format.